src/lib/logger.js

```diff
@@ -19,28 +19,23 @@
 *
 */
 
-import { DefaultLogger } from './DefaultLogger';
-import { ILogger } from '../types';
+import bunyan from 'bunyan';
+import { ILoggerCreate, ILogger } from '../types';
+import path from 'path';
 
-export default class Logger {
-  static _logger: ILogger = DefaultLogger;
-
-  static error(...params: Array<any>) {
-    Logger._logger.error(...params);
-  }
-
-  static info(...params: Array<any>) {
-    Logger._logger.info(...params);
-  }
-
-  static initialize(logger: ILogger) {
-    Logger._logger = logger;
-  }
-
-  static log(...params: Array<any>) {
-    Logger._logger.log(...params);
+export default class Logger implements ILoggerCreate {
+  static createLogger(applicationName: string): ILogger {
+    return bunyan.createLogger({
+      level: process.env.LOG_LEVEL !== undefined ? process.env.LOG_LEVEL : 'info',
+      name: applicationName,
+      serializers: bunyan.stdSerializers,
+    });
   }
-  static warn(...params: Array<any>) {
-    Logger._logger.warn(...params);
+  static createModuleLogger(applicationModule: any): ILogger {
```

Comment:
There is not only ONE Logger, you will get automatically ONE logger per File and can also define own logger if needed.

Later you can filter or select also by these loggerNames

As an result, if you simply type logger.info('HI') , you will still see from which file(logger) this line comes.